instrumentation_test.go

```diff
@@ -178,28 +163,6 @@ func TestWithResourceAttributes(t *testing.T) {
 	})
 }
 
-func TestWithLogLevel(t *testing.T) {
```

Comment:
Do we want to replace these with tests for the new `WithLogger` and also one for the env var `OTEL_LOG_LEVEL`?

Reply:
I'll add a test for `WithLogger`. I'm not sure exactly how to test the CLI for `OTEL_LOG_LEVEL` though.

Reply:
Updated. PTAL.